src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -247,6 +259,9 @@ public void putDataPoint(String metricName,
                     final String value = tags.get(tagName);
                     boolean isCachedValue = tagValueCache.isKnown(value);
                     if (m_cassandraConfiguration.getTagValueCacheSize() > 0 && !isCachedValue) {
+                        if ( span != null) {
```

Comment:
For all these GlobalTracer.get() calls, don't we already have this span as 'span' up above?

Reply:
When kairosdb inserts it's own metrics it somehow bypasses the datapoints JAX call where the top most span is started. Hence adding a tag only if span is present otherwise there were NPE in the logs.